src/test/deploy/functions/cel.spec.ts

```diff
@@ -0,0 +1,530 @@
+import { expect } from "chai";
+import { resolveExpression, ExprParseError } from "../../../deploy/functions/cel";
+import { ParamValue } from "../../../deploy/functions/params";
+
+function stringV(value: string): ParamValue {
+  return new ParamValue(value, false, { string: true, number: false, boolean: false });
+}
+function numberV(value: number): ParamValue {
+  return new ParamValue(value.toString(), false, { string: false, number: true, boolean: false });
+}
+function boolV(value: boolean): ParamValue {
+  return new ParamValue(value.toString(), false, { string: false, number: false, boolean: true });
+}
+
+describe("CEL evaluation", () => {
+  describe("Identity expressions", () => {
+    it("raises when the referenced parameter does not exist", () => {
+      expect(() => {
+        resolveExpression("number", "{{ params.FOO }}", {});
+      }).to.throw(ExprParseError);
+    });
+
+    it("raises when the referenced parameter is of the wrong type", () => {
+      expect(() => {
+        resolveExpression("number", "{{ params.FOO }}", {
+          FOO: boolV(true),
+        });
+      }).to.throw(ExprParseError);
+    });
+
+    it("pulls number parameters", () => {
+      const params: Record<string, ParamValue> = {
+        FOO: numberV(22),
+        BAR: boolV(true),
+        BAZ: new ParamValue("quox", false, { string: true }),
+      };
+
+      expect(resolveExpression("number", "{{ params.FOO }}", params)).to.equal(22);
+      expect(resolveExpression("boolean", "{{ params.BAR }}", params)).to.be.true;
+      expect(resolveExpression("string", "{{ params.BAZ }}", params)).to.equal("quox");
+    });
+  });
+
+  describe("Equality expressions", () => {
+    it("raises when the LHS param does not exist", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ params.FOO == 22 }}", {});
+      }).to.throw(ExprParseError);
+    });
+
+    it("raises when the type of the LHS and RHS do not match", () => {
+      expect(() => {
+        resolveExpression("boolean", "{{ params.FOO == 'asdf' }}", {
+          FOO: numberV(22),
+        });
+      }).to.throw(ExprParseError);
+    });
+
+    it("it determines whether the LHS resolves to the same thing as the RHS", () => {
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == 22 }}", {
+          FOO: numberV(22),
+        })
+      ).to.be.true;
+      expect(
+        resolveExpression("boolean", "{{ params.FOO == 11 }}", {
+          FOO: numberV(22),
+        })
+      ).to.be.false;
+      expect(
+        resolveExpression("boolean", '{{ params.FOO == "bar baz" }}', {
+          FOO: new ParamValue("bar baz", false, { string: true }),
```

Comment:
nit* how is this test different from using `stringV` to define a string param?

Reply:
it isnt